src/cli/util.rs

```diff
@@ -444,21 +444,35 @@ pub fn get_active_cluster<'a>(
     }
 }
 
-pub fn get_username_and_password() -> Result<(String, String), ShellError> {
-    println!("Please enter username:");
-    let username = match read_input() {
-        Some(user) => user,
-        None => {
-            return Err(generic_error("Username required", None, None));
-        }
-    };
-
-    let password = match rpassword::prompt_password("Password: ") {
-        Ok(p) => p,
-        Err(_) => {
-            return Err(generic_error("Password required", None, None));
-        }
-    };
+pub fn get_username_and_password(
+    user_flag: Option<String>,
+    password_flag: Option<String>,
+) -> Result<(String, String), ShellError> {
+    let username = user_flag.map_or_else(
+        || {
+            println!("Please enter username:");
+            read_input().ok_or_else(|| generic_error("Username required", None, None))
+        },
+        Ok,
+    )?;
+
+    let password = password_flag.map_or_else(
+        || match rpassword::prompt_password("Password: ") {
+            Ok(p) => {
+                if p.is_empty() {
```

Comment:
Doesn't look like the original code had this check, is it needed?
I assume there could be cases where we don't want to include a password at all (e.g. when using certs). Does this check get avoided for those cases?

Reply:
The original code didn't have this check as I thought that `prompt_password` returned an error on am empty password, but it does not. 

Even when using certs cbshell needs the username and password to perform any doc ops. Also when we register a cluster through the config file we return an error if there is no username and password, so I think it should be the same when registering a cluster through this command. 

The other place this is used is when creating credentials, where we obviously need a username and password. I think if there was something that used wither certs or username/password it'd be the responsibility of the caller to call this func if they require a username and password, and don't if a cert has been supplied.

Reply:
Thanks for the context, makes sense.